packages/devtools/lib/src/inspector/inspector_tree.dart

```diff
@@ -540,6 +544,74 @@ abstract class InspectorTree {
     });
   }
 
+  String get tooltip;
+  set tooltip(String value);
+
+  RemoteDiagnosticsNode _currentHoverDiagnostic;
+  bool _computingHover = false;
+
+  Future<void> onHover(InspectorTreeNode node, PaintEntry entry) async {
+    if (_onHoverCallback != null) {
+      _onHoverCallback(node, entry?.icon);
+    }
+
+    final diagnostic = node?.diagnostic;
+    final lastHover = _currentHoverDiagnostic;
+    _currentHoverDiagnostic = diagnostic;
+    // Only show tooltips when we are hovering over specific content in a row
+    // rather than over the entire row.
+    // TODO(jacobr): consider showing the tooltip any time we are on a row with
+    // a diagnostics node to make tooltips more discoverable.
+    // To make this work well we would need to add custom tooltip rendering that
+    // more clearly links tooltips to the exact content in a row they apply to.
+    if (diagnostic == null || entry == null) {
+      tooltip = '';
+      _computingHover = false;
+      return;
+    }
+
+    if (entry.icon == defaultIcon) {
+      tooltip = 'Default value';
+      _computingHover = false;
+      return;
+    }
+
+    if (diagnostic.isEnumProperty()) {
+      // We can display a better tooltip than the one provied with the
+      // RemoteDiagnosticsNode as we have access to introspection
+      // via the vm service.
+
+      if (lastHover == diagnostic && _computingHover) {
+        // No need to spam the VMService. We are already computing the hover
+        // for this node.
+        return;
+      }
+      _computingHover = true;
+      // Clear the tooltip while we are computing it.
```

Comment:
What does this comment go to?

Reply:
removed. unfortunately html tooltips are fiddly so I had to remove that or html fails to update the tooltip.